function.k

```diff
@@ -0,0 +1,231 @@
+     <currentAccount> #account(C,N) </currentAccount>
+     <env> ENV:Map  </env>
+     <callStack> .List => ListItem(#state(#account(C,N),ENV,#return(false,#undef_Value), #internal)) ... </callStack> 
+
+rule 
+     <k> #Call(F:Id, Vs:Values) => #BindInputParams(Ps, Vs) ~> #BindReturnParams(OPs, .Values) ~> #CallBody(Ss) ...</k>
+     <currentAccount> #account(C:AccountInt,N:AccountId) </currentAccount>
+     <Contract>
+         <cname> N </cname>
+         <Function>
+             <fname> F </fname>
+             <inputParams> Ps:FParameters </inputParams>
+             <Returns> OPs:FParameters </Returns>
+             <body> Ss:Statements </body>
+             ...
+         </Function>
+         ...
+     </Contract>
+
+rule
+     <k> #BindInputParams(Ps, Vs) => #InitParams(Ps, Vs) ...</k>
+
+rule
+     <k> #BindReturnParams(OPs, Vs) => #InitParams(OPs, Vs) ...</k>
+
+     
+rule 
+     <k> #InitParams(P:FParameter, Ps:FParameters, V:Value,Vs:Values)
+     	    => #InitParam(P, V) ~> #InitParams(Ps, Vs) ...</k>
+
+rule 
+     <k> #InitParams(P:FParameter, Ps:FParameters, .Values)
+     	    => #InitParam(P, #undef_Value) ~> #InitParams(Ps, .Values) ...</k>
+
+rule 
+     <k> #InitParams(.FParameters, .Values) => . ...</k>
+
+rule 
+     <k> #InitParam(T:TypeName S:StorageLocations X:Id, V) => #allocate(C, CN, #varInfo(X:Id, V:Value, T:TypeName, #mem)) ...</k>
+     <currentAccount> #account(C:AccountInt,CN:AccountId) </currentAccount>
+
+rule 
+     <k> #InitParam(T:TypeName, V) => . ...</k>
+
+rule [Call-Function-Body]:
+     <k> #CallBody(S:Statement Ss:Statements) => #exeStmt(S) ~> #CallBody(Ss) ...</k>
+
+rule 
+     <k> #CallBody(.Statements) => . ...</k>
+
+rule
+     <k> #exeStmt(S:SimpleStatement;) => S; ...</k>
+     <callStack> ListItem(#state(#account(_:AccountInt,_:AccountId), _:Map, #return(false,_:Value), _:FunctionFlag)) ... </callStack>
+
+rule
+     <k> #exeStmt(S:SimpleStatement;) => . ...</k>
+     <callStack> ListItem(#state(#account(_:AccountInt,_:AccountId), _:Map, #return(true,_:Value), _:FunctionFlag)) ... </callStack>
+
+rule [Local-Call-Return]:
+     <k> #return  => V  ... </k>
+     <callStack> ListItem(#state(C, Rho, #return(_:Bool,V:Value), #internal)) => .List ... </callStack> 
+     <env> _ => Rho </env>
+
+rule [External-Call-Return]:
+     <k> #returnContract => V ... </k>
+     <callStack> ListItem(#state(C, Rho, #return(_:Bool,V:Value), #external)) => .List ... </callStack>
+     <env> _ => Rho </env>
+     <currentAccount> _ => C  </currentAccount>
+     <callDepth> Depth => Depth -Int 1 </callDepth>
+
+rule [ReturnValue]:
+     <k> return V:Value  => #end_Exp ... </k>
+     <callStack> ListItem(#state(C, Rho, #return(_:Bool => true, _:Value => V), _:FunctionFlag)) ... </callStack>
+
+rule [Return]:
+     <k> return  => #end_Exp ... </k>
+     <callStack> ListItem(#state(C, Rho, #return(_:Bool => true, _:Value), _:FunctionFlag)) ... </callStack>
```

Comment:
I think it would make more sense to refer to the values in the *current* `<callState>` rather than looking at the top of the `<callStack>` for making these decisions.